src/schema-association-service.ts

```diff
@@ -282,25 +271,41 @@ async function autoDetectSchema(
     const authHandler = azdev.getBearerHandler(token.accessToken);
     const azureDevOpsClient = new azdev.WebApi(`https://dev.azure.com/${organizationName}`, authHandler);
 
-    if (vscode.workspace.getConfiguration('azure-pipelines', workspaceFolder).get<boolean>('1ESPipelineTemplatesSchemaFile', false)) {
-        // get cached 1ESPT schema if:
-        // 1. User is signed in with microsoft account
-        // 2. 1ESPT schema is enabled
-        // 3. 1ESPT schema is not older than 24 hours
-        const cachedSchemaUri1ESPT = await getCached1ESPTSchema(context, organizationName, session, lastUpdated1ESPTSchema, seen1ESPTOrganizations);
-        if (cachedSchemaUri1ESPT) {
-            return cachedSchemaUri1ESPT;
+    if (eligibleFor1ESPTIntellisense === undefined) {
+        [eligibleFor1ESPTIntellisense, repoId1espt] = await checkIfUserEligibleFor1ESPTIntellisense(azureDevOpsClient, organizationName);
+    }
+
+    if (eligibleFor1ESPTIntellisense) {
+        // user has enabled 1ESPT schema
+        if (vscode.workspace.getConfiguration('azure-pipelines', workspaceFolder).get<boolean>('1ESPipelineTemplatesSchemaFile', false)) {
+            const cachedSchemaUri1ESPT = await getCached1ESPTSchema(context, organizationName, session, lastUpdated1ESPTSchema, seen1ESPTOrganizations);
+            if (cachedSchemaUri1ESPT) {
+                return cachedSchemaUri1ESPT;
+            }
+            else {
+                // if user is signed in with microsoft account and has enabled 1ESPipeline Template Schema, then give preference to 1ESPT schema
+                const schemaUri1ESPT = await get1ESPTSchemaUri(azureDevOpsClient, organizationName,session, context, repoId1espt);
+                if (schemaUri1ESPT) {
+                    lastUpdated1ESPTSchema.set(organizationName, new Date());
+                    seen1ESPTOrganizations.add(organizationName);
+                    return schemaUri1ESPT;
+                }
+            }
         }
+        // If 1ESPT schema is not enabled, show a pop-up option to enable it for enahnced intellisense
         else {
-            // if user is signed in with microsoft account and has enabled 1ESPipeline Template Schema, then give preference to 1ESPT schema
-            const schemaUri1ESPT = await get1ESPTSchemaUriIfAvailable(azureDevOpsClient, organizationName, session, context, workspaceFolder);
-            if (schemaUri1ESPT) {
-                lastUpdated1ESPTSchema.set(organizationName, new Date());
-                seen1ESPTOrganizations.add(organizationName);
-                return schemaUri1ESPT;
-            }
+            vscode.window.showInformationMessage(Messages.userEligibleForEnahanced1ESPTIntellisense, Messages.enable1ESPTSchema).then(async action => {
```

Comment:
Async/await. Also make sure you have a way to hide this popup (i.e., a "don't ask me again" option that saves to globalState or something) so that people can hide it for good.

Reply:
done